### ISSUE TITLE
Remake GUI with Panel

### ISSUE THREAD

**Issue by martindurant** [ContinuumIO/intake#183]
The GUI is built with ipywidgets. It could be done with Panel, especially if it's to have the ability not to just give basic info about data-sets, but also provide plots and data.

**Comment by martindurant** [comment]
cc @jbednar

**Comment by jbednar** [comment]
Yes, that would be fun.  @philippjfr can speculate about which bits of https://intake.readthedocs.io/en/latest/gui.html would currently give trouble.  It's definitely all something we *should* be able to handle, but I think we don't yet have a decent file requestor, and there may not yet be some other bits needed.  But certainly integration with plots should be easy, including anything where intake has the metadata needed to make .hvplot() work.

**Comment by philippjfr** [comment]
I really like this idea. I haven't yet played with the ipywidgets GUI but will try to do so this weekend to see what it would take to replace it. Together with hvplot you could pretty quickly build a visual data explorer and then load and persist plot specs to the catalogue file.

**Comment by LightnerAndrew** [comment]
Hi @martindurant, I really enjoyed your talk on intake in DC and plan to explore how we can use it at my organization. 

One limitation of using ipywidgets for the GUI for my organization is that some colleagues use Google CoLab, and ipywidgets is not supported in this environment.  Currently, `import intake` causes the runtime to crash.  To isolate the GUI as the cause, I forked intake and omitted the all references to the gui in the __init__.py and was able to successfully import this version of intake in Google CoLab. 

Am I correct that Panel refers to bokeh widgets?  If so, bokeh renders in Google CoLab and thus would be a solution to this issue. 

 If I can make sure all colleagues can use intake, I would also create and contribute a driver for the [Socrata API](https://dev.socrata.com/) where we host the [USAID Development Data Library](data.usaid.gov) and where other public organizations host their data. (I expect I will need a some help  along the way.) 

Thanks for the help and the great work!

**Comment by martindurant** [comment]
> import intake causes the runtime to crash.

That is surprising! It would, at the very least, be useful to avoid that, people should be able to use intake even without any gui. 

> Am I correct that Panel refers to bokeh widgets? 

It is a layer over holoviews, which does indeed render using bokeh

> bokeh renders in Google Colab and thus would be a solution to this issue.

Would be pleased to see it! It would be nice to have multiple gui options for different situations, and planel - which renders both in the notebook and independently - would be great. Furthermore, if the GUI can display a sample of the data or plots, that would make a very nice data browsing standalone app. Just some thoughts...

> I would also create and contribute a driver for the Socrata API

Please do start, and feel free to tag me on issues as you face them.

**Comment by jbednar** [comment]
That all sounds great, but note that Panel is a layer over some parts of *Bokeh*; it does not require or depend on HoloViews.  

In any case, what I'd like to see for Intake is the stack Panel+hvPlot+HoloViews+Bokeh, where a Panel-based GUI can display hvPlot-specified visualizations from Intake catalog files, using Bokeh to render the underlying plots.  I think that combination will make it very easy to set up browsable data catalogs that make it much easier for people to find things of interest to them.

**Comment by martindurant** [comment]
cc @bryevdv

**Comment by martindurant** [comment]
Agreed, Intake should be careful around jupyter-widgets. Funnily, it works if you don't have it installed at all. Do we now which versions cause problems? I guess I can just catch all exceptions at eh point of trying to do GUI stuff.

**Comment by jbednar** [comment]
That would definitely be an improvement, but there may be a way to make intake.gui be a promise or be memoized, lazily populating with something real if someone ever accesses it, but otherwise safely remaining an empty shell that doesn't require even executing any GUI code.
